raiden/tests/integration/transfer/test_mediatedtransfer.py

```diff
@@ -645,7 +648,13 @@ def assert_balances(expected_transferred_amounts=List[int]):
     for i, fee_schedule in enumerate(case.get("incoming_fee_schedules", [])):
         if fee_schedule:
             set_fee_schedule(apps[i + 1], apps[i], fee_schedule)
-    with patch("raiden.routing.get_best_routes_internal", get_best_routes_with_fees):
+
+    route_patch = patch("raiden.routing.get_best_routes_internal", get_best_routes_with_fees)
+    disable_max_mediation_fee_patch = patch(
```

Comment:
@karlb I need some help here. If I disable the max mediation fee check the test passes.

So I naively thought that it's due to the fee being 50% of the amount which is way over that limit.

```
    fee = FeeAmount(5)
    amount = PaymentAmount(10)
```

So I tried to change it (both with the limit on and off) and the test still did not pass but failed in very weird ways.

So I am kind of confused why. I assumed that no matter what (sensible) values I give to the `amount` and `fee` the test should pass. Any ideas?

Reply:
Apart from potentially different rounding, the tests should work with arbitrary values (I haven't tested it, though). I'll look through this PR and if I don't find any good explanation, I'll check out the branch and debug.

Reply:
From what I've seen so far (tests are slow, so I can't try many things), everything works fine as long as the default node deposit of 200 is sufficient and fails in non-obvious ways when going above that.

Reply:
Just now I also managed to get a timeout in a case where the deposit is not the limit. I'm looking into it.

Reply:
OK, three things that can make the tests failed when you change the numbers:
* fee too small, which prevents the unlock message from being sent
* transfers exceed default deposit of 200
* increasing the fee amount so that DEFAULT_MEDIATION_FEE_MARGIN is not rounded away, anymore.

At least the last point should be handled somehow, since it's ugly to have tests failing in complicated ways when we change that value.

Reply:
See https://github.com/raiden-network/raiden/pull/4786. Do you think we need any more changes, here?